src/main/kotlin/org/vividus/testsite/api/HttpRequestController.kt

```diff
@@ -0,0 +1,31 @@
+package org.vividus.testsite.api
+
+import com.github.benmanes.caffeine.cache.Caffeine
+import com.github.benmanes.caffeine.cache.LoadingCache
+import java.util.UUID
+import java.util.concurrent.TimeUnit
+import org.springframework.http.ResponseEntity
+import org.springframework.web.bind.annotation.GetMapping
+import org.springframework.web.bind.annotation.RequestParam
+import org.springframework.web.bind.annotation.RestController
+
+@RestController
+class HttpRequestController {
+    val cache: LoadingCache<UUID, Long> = Caffeine.newBuilder()
+            .expireAfterWrite(10, TimeUnit.SECONDS)
+            .build { System.currentTimeMillis() }
+
+    @GetMapping("/api/delayed-response")
+    fun delayResponse(@RequestParam clientId: UUID): ResponseEntity<Unit> {
+        val firstRequestTimestamp = cache.get(clientId)
+        if (firstRequestTimestamp != null && countAgeInSeconds(firstRequestTimestamp) < 5) {
```

Comment:
when can `firstRequestTimestamp` be `null` having current implementation?

Reply:
cache.get (clientId) return not Long, but Long? and function countAgeInSeconds (firstRequestTimestamp: Long): Long won't take this value untill we checked it's not Null